source/DSCResources/MSFT_ADFineGrainedPasswordPolicy/MSFT_ADFineGrainedPasswordPolicy.psm1

```diff
@@ -0,0 +1,802 @@
+                @getADFineGrainedPasswordPolicySubjectParams).Name
+        }
+        catch
+        {
+            $errorMessage = $script:localizedData.RetrieveFineGrainedPasswordPolicySubjectError -f $Name
+            New-InvalidOperationException -Message $errorMessage -ErrorRecord $_
+        }
+    }
+
+    if ($policy)
+    {
+        return @{
+            Name                        = $Name
+            ComplexityEnabled           = $policy.ComplexityEnabled
+            LockoutDuration             = $policy.LockoutDuration
+            LockoutObservationWindow    = $policy.LockoutObservationWindow
+            LockoutThreshold            = $policy.LockoutThreshold
+            MinPasswordAge              = $policy.MinPasswordAge
+            MaxPasswordAge              = $policy.MaxPasswordAge
+            MinPasswordLength           = $policy.MinPasswordLength
+            PasswordHistoryCount        = $policy.PasswordHistoryCount
+            ReversibleEncryptionEnabled = $policy.ReversibleEncryptionEnabled
+            Precedence                  = $policy.Precedence
+            Ensure                      = 'Present'
+            Subjects                    = $policySubjects
+        }
+    }
+    else
+    {
+        return @{
+            Name                        = $Name
+            ComplexityEnabled           = $null
+            LockoutDuration             = $null
+            LockoutObservationWindow    = $null
+            LockoutThreshold            = $null
+            MinPasswordAge              = $null
+            MaxPasswordAge              = $null
+            MinPasswordLength           = $null
+            PasswordHistoryCount        = $null
+            ReversibleEncryptionEnabled = $null
+            Precedence                  = $null
+            Ensure                      = 'Absent'
+            Subjects                    = $policySubjects
+        }
+    }
+} #end Get-TargetResource
+
+<#
+    .SYNOPSIS
+        Determines if the Active Directory fine-grained password policy is in
+        the desired state
+
+    .PARAMETER Name
+        Specifies an Active Directory fine-grained password policy object name.
+
+    .PARAMETER Precedence
+        Specifies a value that defines the precedence of a fine-grained password policy among all fine-grained password
+        policies.
+
+    .PARAMETER DisplayName
+        Specifies the display name of the object.
+
+    .PARAMETER Subjects
+        Specifies the ADPrincipal names the policy is to be applied to, overwrites all existing.
+
+    .PARAMETER Ensure
+        Specifies whether the fine grained password policy should be present or absent. Default value is 'Present'.
+
+    .PARAMETER ComplexityEnabled
+        Specifies whether password complexity is enabled for the password policy.
+
+    .PARAMETER LockoutDuration
+        Specifies the length of time that an account is locked after the number of failed login attempts exceeds the
```

Comment:
Please adjust the `LockoutDuration`, `LockOutObservationWindow`, `MinPasswordAge` and `MaxPasswordAge` descriptions to the new ones specified in the schema in both the `Test` and `Set` functions.